File: software/examples/java/ExampleCallback.java
import com.tinkerforge.IPConnection;
import com.tinkerforge.BrickletColor;

public class ExampleCallback {
	private static final String HOST = "localhost";
	private static final int PORT = 4223;

	// Change XYZ to the UID of your Color Bricklet
	private static final String UID = "XYZ";

	// Note: To make the example code cleaner we do not handle exceptions. Exceptions
	//       you might normally want to catch are described in the documentation
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletColor c = new BrickletColor(UID, ipcon); // Create device object

		ipcon.connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Add color listener
		c.addColorListener(new BrickletColor.ColorListener() {
			public void color(int r, int g, int b, int c) {
				System.out.println("Color [R]: " + r);
				System.out.println("Color [G]: " + g);
				System.out.println("Color [B]: " + b);
				System.out.println("Color [C]: " + c);
				System.out.println("");
			}
		});

		// Set period for color callback to 1s (1000ms)
		// Note: The color callback is only called every second
		//       if the color has changed since the last call!
		c.setColorCallbackPeriod(1000);

		System.out.println("Press key to exit"); System.in.read();
		ipcon.disconnect();
	}
}

File: software/examples/php/ExampleCallback.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletColor.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletColor;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change to your UID

// Callback function for color callback
function cb_color($r, $g, $b, $c)
{
    echo "Color[R]: $r\n";
    echo "Color[G]: $g\n";
    echo "Color[B]: $b\n";
    echo "Color[C]: $c\n";
    echo "\n";
}

$ipcon = new IPConnection(); // Create IP connection
$c = new BrickletColor(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Register color callback to function cb_color
$c->registerCallback(BrickletColor::CALLBACK_COLOR, 'cb_color');

// Set period for color callback to 1s (1000ms)
// Note: The color callback is only called every second
//       if the color has changed since the last call!
$c->setColorCallbackPeriod(1000);

echo "Press ctrl+c to exit\n";
$ipcon->dispatchCallbacks(-1); // Dispatch callbacks forever

?>
